pkg/cli/app.go

```diff
@@ -113,6 +115,8 @@ func (app *App) RunContext(ctx context.Context, arguments []string) (err error)
 		args := Args(parentCtx.Args().Slice())
 		ctx := newContext(parentCtx.Context, app)
 
+		defer engine.Shutdown(ctx)
```

Comment:
This is not the right place for this. Any pacakges in the `/pkg` folder should be shared libraries (unaware of Terragrunt) that we should be able to move anywhere, like `go-commons` over time. Could you consider moving it to:
https://github.com/gruntwork-io/terragrunt/blob/ec2c1ebbd2767c9bc1a70aa2d4266ca7d4b86887/cli/app.go#L93
or 
https://github.com/gruntwork-io/terragrunt/blob/ec2c1ebbd2767c9bc1a70aa2d4266ca7d4b86887/cli/app.go#L190